k9mail/src/main/java/com/fsck/k9/mailstore/StoreSchemaDefinition.java

```diff
@@ -51,556 +51,323 @@ private void upgradeDatabase(final SQLiteDatabase db) {
-                db.execSQL("DROP INDEX IF EXISTS msg_folder_id");
-                db.execSQL("DROP INDEX IF EXISTS msg_folder_id_date");
-                db.execSQL("CREATE INDEX IF NOT EXISTS msg_folder_id_deleted_date ON messages (folder_id,deleted,internal_date)");
-
-                db.execSQL("DROP INDEX IF EXISTS msg_empty");
-                db.execSQL("CREATE INDEX IF NOT EXISTS msg_empty ON messages (empty)");
-
-                db.execSQL("DROP INDEX IF EXISTS msg_read");
-                db.execSQL("CREATE INDEX IF NOT EXISTS msg_read ON messages (read)");
-
-                db.execSQL("DROP INDEX IF EXISTS msg_flagged");
-                db.execSQL("CREATE INDEX IF NOT EXISTS msg_flagged ON messages (flagged)");
-
-                db.execSQL("DROP INDEX IF EXISTS msg_composite");
-                db.execSQL("CREATE INDEX IF NOT EXISTS msg_composite ON messages (deleted, empty,folder_id,flagged,read)");
-
-
-
-                db.execSQL("DROP TABLE IF EXISTS threads");
-                db.execSQL("CREATE TABLE threads (" +
-                        "id INTEGER PRIMARY KEY, " +
-                        "message_id INTEGER, " +
-                        "root INTEGER, " +
-                        "parent INTEGER" +
-                        ")");
-
-                db.execSQL("DROP INDEX IF EXISTS threads_message_id");
-                db.execSQL("CREATE INDEX IF NOT EXISTS threads_message_id ON threads (message_id)");
-
-                db.execSQL("DROP INDEX IF EXISTS threads_root");
-                db.execSQL("CREATE INDEX IF NOT EXISTS threads_root ON threads (root)");
-
-                db.execSQL("DROP INDEX IF EXISTS threads_parent");
-                db.execSQL("CREATE INDEX IF NOT EXISTS threads_parent ON threads (parent)");
-
-                db.execSQL("DROP TRIGGER IF EXISTS set_thread_root");
-                db.execSQL("CREATE TRIGGER set_thread_root " +
-                        "AFTER INSERT ON threads " +
-                        "BEGIN " +
-                        "UPDATE threads SET root=id WHERE root IS NULL AND ROWID = NEW.ROWID; " +
-                        "END");
-
-                db.execSQL("DROP TABLE IF EXISTS pending_commands");
-                db.execSQL("CREATE TABLE pending_commands " +
-                           "(id INTEGER PRIMARY KEY, command TEXT, arguments TEXT)");
-
-                db.execSQL("DROP TRIGGER IF EXISTS delete_folder");
-                db.execSQL("CREATE TRIGGER delete_folder BEFORE DELETE ON folders BEGIN DELETE FROM messages WHERE old.id = folder_id; END;");
-
-                db.execSQL("DROP TRIGGER IF EXISTS delete_message");
-                db.execSQL("CREATE TRIGGER delete_message " +
-                        "BEFORE DELETE ON messages " +
-                        "BEGIN " +
-                        "DELETE FROM message_parts WHERE root = OLD.message_part_id;" +
-                        "END");
+                dbLegacyRecreateDatabase(db);
```

Comment:
This path is also taken when a new database is created. So we should find a better name for this method.